pulumi/volumeReaper/src/volumeReaper.ts

```diff
@@ -127,12 +134,10 @@ export class VolumeReaper {
       undefined,
       `statefulset=${this.name}`
     )
-
-    const items = snapshots.body.items
+    const items = snapshots.body.items.map(this.deserialize);
     console.log(`Found ${items.length} snapshots for ${this.namespace}.${this.name}`)
-
     // sorted newest -> oldest
-    return items.sort((a, b) => b.metadata.creationTimestamp.getTime() - a.metadata.creationTimestamp.getTime())
+    return items.sort((a, b) => new Date(b.metadata.creationTimestamp).getTime() - new Date(a.metadata.creationTimestamp).getTime())
```

Comment:
now we are deserializing
```suggestion
    return items.sort((a, b) => b.metadata.creationTimestamp.getTime() - a.metadata.creationTimestamp.getTime())
```